src/object-diagram/logic/reader/panelViewInputObjectDiagramReader.ts

```diff
@@ -1,15 +1,23 @@
-import { PanelViewInputVariableMap, VariableRelation } from '../../../model/panelViewInput';
+import { hasNullPrefix, hasObjectPrefix, hasVariablePrefix } from '../../../util/nodePrefixHandler';
+import { PanelViewInputVariableMap, PanelViewVariable, PrimitiveValue, VariableRelation } from '../../../model/panelViewInput';
 import { escapeString } from '../../model/escapedString';
 import { Field } from '../../model/field';
 import { ObjectDiagram } from '../../model/objectDiagram';
-import { isPrimitiveJavaType, PrimitiveJavaType } from '../../model/primitiveJavaType';
 import { Reference } from '../../model/reference';
 import { Structure } from '../../model/structure';
-import { createStructureId } from '../../model/structureId';
+import { createStructureId, StructureId } from '../../model/structureId';
 import { ObjectDiagramReader } from './objectDiagramReader';
 
-const isUndefinedOrPrimitiveJavaType = (type: string | undefined): type is undefined | PrimitiveJavaType =>
-  type === undefined || isPrimitiveJavaType(type);
+const isObject = (variable: PanelViewVariable): boolean => hasObjectPrefix(variable.id);
+
+const isNullOfObject = (variable: PanelViewVariable): boolean =>
+  hasNullPrefix(variable.id) && variable.incomingRelations !== undefined && Boolean(variable.incomingRelations[0].relationName);
```

Comment:
Does this function test for `null` references coming from an object?

Reply:
Yes.